### PR TITLE
coreRT fat pointer hidden argument fix for calls with retBufArg

### PR DESCRIPTION
**Pull request by sandreenko**
in case of 
`class A<T>
{
	BigStruct foo(int arg1);
}`

we expect 
`void foo(this, retBufPointer, hiddenDictionary, arg1)` for amd64 and arm.
`void foo(this, retBufPointer, arg1,  hiddenDictionary)` for x86.
Fix dotnet/corert/issues/2751.

### CODE REVIEW

**Comment by sandreenko** [comment]
PTAL @BruceForstall 
cc @MichalStrehovsky  @dotnet/jit-contrib

**Comment by sandreenko** [comment]
WIP. I need to check platform and var arg cookie position.

**Comment by sandreenko** [comment]
@BruceForstall please look again.

File: src/jit/flowgraph.cpp [hunk 1]
```diff
@@ -24495,13 +24495,58 @@ class FatCalliTransformer
...
+        //
+        // Arguments:
+        //    argList - fat call node
+        //    hiddenArgument - generic context hidden argument
+        //
+        void AddArgumentToTail(GenTreeArgList* argList, GenTreePtr hiddenArgument)
+        {
+            GenTreeArgList* iterator = argList;
+            while (iterator->Rest())
+            {
```
Comment (BruceForstall) [inline src/jit/flowgraph.cpp]: nit:
```
            while (iterator->Rest() != nullptr)
```

File: src/jit/flowgraph.cpp [hunk 2]
```diff
@@ -24495,13 +24495,58 @@ class FatCalliTransformer
...
+                newArgs                   = compiler->gtNewListNode(retBuffer, newArgs);
+            }
+            else
+            {
+                newArgs = compiler->gtNewListNode(hiddenArgument, oldArgs);
+            }
+#else
+            newArgs = oldArgs;
+            AddArgumentToTail(newArgs, hiddenArgument);
+#endif
```
Comment (BruceForstall) [inline src/jit/flowgraph.cpp]: In this case (x86), is it possible to have a var args cookie? If so, the var args cookie comes last, after the generics context. (I can't remember if you can have a generic function that is also var args)

Comment (sandreenko) [inline src/jit/flowgraph.cpp]: No, we don't have such functions. There are several asserts that check it.

**Comment by BruceForstall** [comment]
LGTM, assuming the answer to my var args question is that it's not a possibility for this case.

**Pull request merged**